src/runtime/Object.cpp

```diff
@@ -1655,100 +1657,149 @@ bool Object::setIndexedProperty(ExecutionState& state, const Value& property, co
     return set(state, ObjectPropertyName(state, property), value, receiver);
 }
 
-void Object::privateFieldAdd(ExecutionState& state, AtomicString propertyName, const Value& value)
+static ObjectPrivateMemberDataChain* ensurePieceOnPrivateMemberChain(ExecutionState& state, ObjectExtendedExtraData* e, Object* contextObject)
 {
-    auto& v = ensureObjectExtendedExtraData()->m_privateFieldValues;
-    for (size_t i = 0; i < v.size(); i++) {
-        if (UNLIKELY(v[i].first == propertyName)) {
-            ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Cannot add private field %s with same name twice", propertyName.string());
+    Optional<ObjectPrivateMemberDataChain*> piece = e->m_privateMemberChain;
+    if (e->m_privateMemberChain) {
+        while (true) {
```

Comment:
How about re-write a while statement using the expression below?

```suggestion
        while (piece) {
```

Reply:
piece is always true.